bash_completion

```diff
@@ -158,10 +158,31 @@ quote_readline()
     printf %s "$ret"
 } # quote_readline()
 
+# shellcheck disable=SC1003
+_comp_dequote__initialize()
+{
+    unset -f "$FUNCNAME"
+    local param='\$([_a-zA-Z][_a-zA-Z0-9]*|[-*@#?$!0-9_])|\$\{[!#]?([_a-zA-Z][_a-zA-Z0-9]*(\[([0-9]+|[*@])\])?|[-*@#?$!0-9_])\}'
+    local quote='\\.|'\''[^'\'']*'\''|"([^\"$`!]|'$param'|\\.)*"|\$'\''([^\'\'']|\\.)*'\'''
+    _comp_dequote__regex_safe_word='^([^\'\''"$`;&|<>()!]|'$quote'|'$param')*$'
+}
+_comp_dequote__initialize
+
+_comp_dequote()
+{
+    ret=() # fallback value for unsafe word and failglob
+    [[ $1 =~ $_comp_dequote__regex_safe_word ]] || return 1
+    eval "ret=($1)" 2>/dev/null # may produce failglob
+}
+
 # This function shell-dequotes the argument
 dequote()
```

Comment:
I'd prefer to deprecate `dequote` and `quote`, and remove their uses from the codebase altogether (at least one `quote` instance remains in `_export`). They are in the shell's global command namespace, look like commands the users are supposed to execute interactively, while never having been intended as such I believe.

Reply:
- I have marked `dequote` with `@deprecated` bf77c2f1.
- I have also added `_comp_quote` that is supposed replace `quote` and marked `quote` with `@deprecated` in 6b2b97e1.
- I have updated the uses of `quote` in `_reply_compgen_array` (`completions/_umount.linux`) and `_export` (`completions/export`) in 1cd1d671. Maybe I could have updated these functions so that they do not perform any quoting, but I have preserved the current behavior.